addons/account/tests/common.py

```diff
@@ -386,6 +386,7 @@ def init_invoice(cls, move_type, partner=None, invoice_date=None, post=False, pr
         if not move_form._get_modifier('date', 'invisible'):
             move_form.date = move_form.invoice_date
         move_form.partner_id = partner or cls.partner_a
+        move_form.currency_id = currency if currency else cls.company_data['currency']
```

Comment:
Maybe this is the line making the test fail.

Try to launch those tests locally and see if those fail.
If it does, then try to only set this if there is a currency provided. (instead of setting to None when nothing is provided)